### DISCUSSION TITLE
OpenAM Docker

### DISCUSSION THREAD

**Discussion by neel-quantasis** [OpenIdentityPlatform/OpenAM#822] [Q&A] [answered]
I've created a docker instance using this documentation ([link](https://github.com/OpenIdentityPlatform/OpenAM/wiki/How-to-Setup-SAMLv2-Federation-in-OpenAM))

How do I make sure the data (configuration setup, idp, users) is not wiped out if the container restarts?

Also, how do I enable CORS?

**Comment by maximthomas** [comment]
> How do I make sure the data (configuration setup, idp, users) is not wiped out if the container restarts?

You need to mount `/usr/openam/config` directory in the Docker container as a volume. More details [here](https://github.com/OpenIdentityPlatform/OpenAM/wiki/TIP:-How-to-Persist-OpenAM-Docker-Container-Data-Between-Restarts)

> Also, how do I enable CORS?

Check this [TIP: How to Enable CORS](https://github.com/OpenIdentityPlatform/OpenAM/wiki/TIP%3A-How-to-Enable-CORS)

**Reply by neel-quantasis** [reply]
@maximthomas 

I was able to persist the user I created after a container restart. I will confirm persistence about IDP and SP.

About CORS,
I followed the steps given on that link
I'm still getting CORS error

I'm using this tool to check whether CORS is enabled or not
https://cors-test.codehappy.dev/?url=http%3A%2F%2Fidp.convertcurrency.online%3A8080%2Fopenam%2F&origin=http%3A%2F%2Flocalhost%3A4200&method=post

Do I have to restart the container after I'm enabling CORS?

**Reply by maximthomas** [reply]
> Do I have to restart the container after I'm enabling CORS?

Yes, restart the container

**Reply by neel-quantasis** [reply]
@maximthomas 

setting a persistent volume on a linux environment gives following output

```bash
docker exec -w '/usr/openam/ssoconfiguratortools' openam-idp bash -c \
'echo "ACCEPT_LICENSES=true
SERVER_URL=http://idp.convertcurrency.online:8080
DEPLOYMENT_URI=/$OPENAM_PATH
BASE_DIR=$OPENAM_DATA_DIR
locale=en_US
PLATFORM_LOCALE=en_US
AM_ENC_KEY=
ADMIN_PWD=passw0rd
AMLDAPUSERPASSWD=p@passw0rd
COOKIE_DOMAIN=idp.convertcurrency.online
ACCEPT_LICENSES=true
DATA_STORE=embedded
DIRECTORY_SSL=SIMPLE
DIRECTORY_SERVER=idp.convertcurrency.online
DIRECTORY_PORT=50389
DIRECTORY_ADMIN_PORT=4444
DIRECTORY_JMX_PORT=1689
ROOT_SUFFIX=dc=openam,dc=example,dc=org
DS_DIRMGRDN=cn=Directory Manager
DS_DIRMGRPASSWD=passw0rd" > conf.file && java -jar openam-configurator-tool*.jar --file conf.file'
12/12/2024 11:54:37:595 AM UTC: Checking license acceptance...
12/12/2024 11:54:37:595 AM UTC: License terms accepted.
12/12/2024 11:54:37:599 AM UTC: Checking configuration directory /usr/openam/config.
12/12/2024 11:54:37:599 AM UTC: ...Success.
12/12/2024 11:54:37:599 AM UTC: Installing OpenAM configuration store
12/12/2024 11:54:37:600 AM UTC: ...Success RSA/ECB/OAEPWithSHA1AndMGF1Padding.
12/12/2024 11:54:37:601 AM UTC: Extracting OpenDJ, please wait...
Configuration Failed.  The server returned error code :500
```

**Reply by neel-quantasis** [reply]
@maximthomas 

I was able to complete the setup using named volumes

After I enable CORS and restart the container, I'm still getting CORS error

**Reply by maximthomas** [reply]
Please attach your CORS settings screenshot and `curl` CORS request output in verbose mode or HAR log

**Reply by neel-quantasis** [reply]
Here you go @maximthomas 

![image](https://github.com/user-attachments/assets/7429d07d-547d-470f-b26e-dd0e8a79261c)

[localhost.har.txt](https://github.com/user-attachments/files/18111772/localhost.har.txt)

**Reply by maximthomas** [reply, marked as answer]
There is no Origin header in your HTTP request.

**Reply by neel-quantasis** [reply]
@maximthomas This works! Thanks!

**Reply by neel-quantasis** [reply]
@maximthomas can you update the documentation for CORS and persistent docker volumes?

**Reply by maximthomas** [reply]
Yes, sure
